src/main/kotlin/org/samberry/recentorder/OrderTimeline.kt

```diff
@@ -0,0 +1,61 @@
+package org.samberry.recentorder
+
+import org.springframework.stereotype.Service
+import java.util.concurrent.Callable
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.Executors
+
+private const val NUMBER_OF_WORKERS = 10
+private const val MILLISECONDS_IN_SECOND = 1000
+private const val TOTAL_NUMBER_OF_TIMESTAMPS = ORDER_DURATION_SECONDS * MILLISECONDS_IN_SECOND
+private const val TIMESTAMPS_PER_WORKER = TOTAL_NUMBER_OF_TIMESTAMPS.div(NUMBER_OF_WORKERS)
+
+@Service
+class OrderTimeline {
+    private val timeline: MutableMap<OrderTimestamp, MutableOrderStatistics> = ConcurrentHashMap()
```

Comment:
Initially immutable objects were used as values. Moving the update synchronization to the object level instead of requiring ConcurrentHashMap to enforce it on put provided a significant performance increase. My naive brain would assume it to be the same, but for whatever reason it is not. Must be an implementation detail of ConcurrentHashMap.